--- jquery.dataTables-1.6.1.js	Sat Jan 30 12:13:22 2010
+++ jquery.dataTables.fixed.js	Sat Feb 06 02:49:26 2010
@@ -561,7 +561,17 @@
 		/*
 		 * text sorting
 		 */
-		"string-asc": function ( a, b )
+		"undefined-asc": function ( a, b )
+        {
+                return 0;
+        },
+        
+        "undefined-desc": function ( a, b )
+        {
+                return 0;
+        },
+        
+        "string-asc": function ( a, b )
 		{
 			var x = a.toLowerCase();
 			var y = b.toLowerCase();
@@ -637,16 +647,18 @@
 		 */
 		"numeric-asc": function ( a, b )
 		{
-			var x = a == "-" ? 0 : a;
-			var y = b == "-" ? 0 : b;
-			return x - y;
+			//var x = a == "-" ? 0 : a;
+			//var y = b == "-" ? 0 : b;
+			//return x - y;
+            return a - b;
 		},
 		
 		"numeric-desc": function ( a, b )
 		{
-			var x = a == "-" ? 0 : a;
-			var y = b == "-" ? 0 : b;
-			return y - x;
+			//var x = a == "-" ? 0 : a;
+			//var y = b == "-" ? 0 : b;
+			//return y - x;
+            return b - a;
 		}
 	};
 	
